recipes/sdl/all/conandata.yml

```diff
@@ -2,3 +2,7 @@ sources:
   "2.0.14":
     url: "https://www.libsdl.org/release/SDL2-2.0.14.tar.gz"
     sha256: d8215b571a581be1332d2106f8036fcb03d12a70bae01e20f424976d275432bc
+patches:
+  "2.0.14":
+    - patch_file: "patches/2.0.14-0001-don-t-build-hidapi-as-shared.patch"
```

Comment:
Upstream is pretty welcome of patches, so after you have checked out if this works, please try to upstream a solution